test/NHSD.BuyingCatalogue.Documents.API.Integration.Tests/Support/AzureBlobStorageScenarioContext.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using FluentAssertions;
+
+namespace NHSD.BuyingCatalogue.Documents.API.IntegrationTests.Support
+{
+    internal class AzureBlobStorageScenarioContext
+    {
+        private const string ConnectionString = @"UseDevelopmentStorage=true";
+        private const string ContainerName = "container-1";
+        private const string SampleDataPath = "SampleData";
+        public Dictionary<string, string> SolutionIdsToGuids = new Dictionary<string, string>();
+        private readonly BlobContainerClient _blobContainer;
+
+        public AzureBlobStorageScenarioContext()
+        {
+            var blobServiceClient = new BlobServiceClient(ConnectionString);
+            _blobContainer = blobServiceClient.GetBlobContainerClient(ContainerName);
+        }
+
+        public async Task InsertFileToStorage(string solutionId, string fileName)
+        {
+            InsertIntoMapping(solutionId);
+            var blobClient = _blobContainer.GetBlobClient(Path.Combine(SolutionIdsToGuids[solutionId], fileName));
+            using var uploadFileStream = File.OpenRead(Path.Combine(SampleDataPath, solutionId, fileName));
+            var response = await blobClient
+                .UploadAsync(uploadFileStream, new BlobHttpHeaders())
+                .ConfigureAwait(false);
+
+            response.GetRawResponse().Status.Should().Be(201);
+        }
+
+        public async Task ClearStorage()
+        {
+            foreach (var directory in SolutionIdsToGuids.Values)
+            {
+                foreach (var blob in _blobContainer.GetBlobs(prefix: directory))
+                {
+                    await _blobContainer.DeleteBlobAsync(blob.Name);
+                }
+            }
+        }
+
+        public string TryToGetGuidFromSolutionId(string solutionId)
+        {
+            if (!SolutionIdsToGuids.TryGetValue(solutionId, out string slnId))
+            {
```

Comment:
Is a `TryGetValue` necessary as you map solution IDs to a `Guid` in your **Given**?

If a solution ID doesn't have a corresponding `Guid` does it not imply that something has gone wrong with the test setup and therefore it might be better to allow a `KeyNotFoundException` to be thrown?

Reply:
Good suggestion, however it is necessary as TryGetValue allows for introducing test cases with invalid solution (solution that doesn't exist). The scenario #1, in its last test case, asserts that when we query for 'sol99' it returns an empty list. Should we remove this and let it throw, it would always throw and never hit the endpoint.

Reply:
Ah, I missed that. Thanks for clarifying.

When a solution ID doesn't exist might it make sense to return `Guid.Empty.ToString()` so that test cases such as this follow the same convention (i.e. using a `Guid` in place of the solution ID) as the test cases where the solution ID does exist?

Also, is there any way of improving the variable names to help distinguish between `solutionId` and `slnId` without resorting to Hungarian-style naming?

Reply:
Good idea! Resorted to solutionIdAsGuid for now. Hope that doesn't make it too confusing.